### ISSUE TITLE
Sprout does not honour integrity protection when deployed without an I-CSCF

### ISSUE THREAD

**Issue by AlexHockey** [Metaswitch/sprout#1451]
#### Symptoms

Clearwater-sip-stress scenarios fail when running against a deployment that does not use an S-CSCF. They fail at the re-registration stage. 
#### Impact
- Calls made by the stress package fail.
- Unecessary re-authentication flows. 
- Tests made by 3rd party verification tools may fail.
#### Release and environment

release-98
#### Steps to reproduce

Set up a deployment with latest chef and release-98 packages (or alternatively, latest chef and packages, but using `--scscf-only`). Set up SIP stress. Monitor the SIPp output and wait 10 mins. The re-registration flows will all fail.

**Comment by AlexHockey** [comment]
The failures are caused because this REGISTER is challenged with a 401:

```
REGISTER sip:2010000000@ajh2.cw-ngv.com SIP/2.0
Via: SIP/2.0/TCP 10.9.182.172:50458;rport;branch=z9hG4bKPjPlxMhebbrzfwOCNWezI1Ym6jefM2-3E2
Path: <sip:byN1DsIln/@10.9.182.172:5058;transport=TCP;lr;ob>
Via: SIP/2.0/TCP 10.81.150.198:5060;rport=34716;received=54.226.86.133;branch=z9hG4bK-14553-1-11-2010000000-4.000000
Max-Forwards: 70
From: <sip:2010000000@ajh2.cw-ngv.com>;tag=14553SIPpTag001
To: <sip:2010000000@ajh2.cw-ngv.com>
Call-ID: 2010000000///1-14553@10.81.150.198
CSeq: 5 REGISTER
User-Agent: Accession 4.0.0.0
Supported: outbound, path
Contact: <sip:2010000000@54.226.86.133:34716;transport=TCP;ob>;+sip.ice;reg-id=1;+sip.instance="<urn:uuid:00000000-0000-0000-0000-000000000001>"
Contact: <sip:2010000000@10.81.150.198:5060;transport=TCP;ob>;expires=0;+sip.ice;reg-id=1;+sip.instance="<urn:uuid:00000000-0000-0000-0000-000000000001>"
Expires: 3600
Allow: PRACK, INVITE, ACK, BYE, CANCEL, UPDATE, SUBSCRIBE, NOTIFY, REFER, MESSAGE, OPTIONS
Authorization: Digest response="", username="2010000000@ajh2.cw-ngv.com", nonce="",integrity-protected=ip-assoc-yes
P-Visited-Network-ID: ajh2.cw-ngv.com
P-Charging-Function-Addresses: ccf=cdf.ajh2.cw-ngv.com
P-Charging-Vector: icid-value="2010000000///1-1455310.81.150.198";icid-generated-at=ec2-54-147-167-201.compute-1.amazonaws.com
Route: <sip:scscf.sprout.ajh2.cw-ngv.com:5054;transport=TCP;lr;orig>
Content-Length:  0
```

The `integrity-protected` parameter is set to `ip-assoc-yes` so this should not have been challenged.

**Comment by AlexHockey** [comment]
Note this problem doesn't occur when sprout is acting as an I-CSCF.

**Comment by richardwhiuk** [comment]
We fail the check on https://github.com/Metaswitch/sprout/blob/faf5e77dca0c7695bf4868908b94539e3c824321/src/authentication.cpp#L714

as the next routing header is: `sip:scscf.sprout.rjw2.cw-ngv.com:5054;transport=TCP;lr;orig`

whereas the scscf_uri is: `sip:scscf.sprout.rjw2.cw-ngv.com:5054;transport=TCP`

**Comment by rkd-msw** [comment]
Fixed by #1565.
